### PR TITLE
feat(global-config): Add support for options in global config

### PR DESCRIPTION
**Pull request by Dav1dde**
Sentry PR: https://github.com/getsentry/sentry/pull/61024

#skip-changelog

### CODE REVIEW

File: relay-server/src/actors/global_config.rs [hunk 1]
```diff
@@ -38,11 +38,9 @@ type UpstreamQueryResult =
 #[derive(Debug, Deserialize, Serialize)]
 #[serde(rename_all = "camelCase")]
 struct GetGlobalConfigResponse {
-    #[serde(default)]
```
Comment (Dav1dde) [inline relay-server/src/actors/global_config.rs:41]: This is redundant and just cleanup

File: relay-dynamic-config/src/global.rs [hunk 2]
```diff
@@ -14,7 +15,11 @@ use serde::{Deserialize, Serialize};
 #[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
 pub struct GlobalConfig {
     /// Configuration for measurements normalization.
+    #[serde(skip_serializing_if = "Option::is_none")]
     pub measurements: Option<MeasurementsConfig>,
+    /// Sentry options passed down to relay.
```
Comment (olksdr) [inline relay-dynamic-config/src/global.rs]:
```suggestion
    /// Sentry options passed down to Relay.
```

File: relay-dynamic-config/src/global.rs [hunk 3]
```diff
@@ -1,9 +1,10 @@
-use std::fs::File;
 use std::io::BufReader;
 use std::path::Path;
+use std::{collections::HashMap, fs::File};
```
Comment (olksdr) [inline relay-dynamic-config/src/global.rs]: Let's split it into the separate imports.

**Review by olksdr — APPROVED** [review]

File: relay-dynamic-config/src/global.rs [hunk 4]
```diff
@@ -34,13 +40,27 @@ impl GlobalConfig {
...
+#[cfg_attr(feature = "jsonschema", derive(JsonSchema))]
+pub struct Options {
+    // Example:
+    // ```rs
+    // #[serde(default, rename = "relay.some-option.name")]
+    // pub some_option: Vec<u32>,
+    // ```
+    /// All other unknown options.
+    #[serde(flatten)]
+    other: HashMap<String, Value>,
```
Comment (iker-barriocanal) [inline relay-dynamic-config/src/global.rs:55]: nit: I assume this `other` is for backward compatibility -- does using `RawValue` avoid deserializing the options we don't need? Haven't checked how serde deals with this exactly.

**Pull request merged**